{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "f4294ca1",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Registered S3 method overwritten by 'spatstat':\n",
      "  method     from\n",
      "  print.boxx cli \n",
      "\n",
      "Attaching SeuratObject\n",
      "\n",
      "Registered S3 method overwritten by 'SeuratDisk':\n",
      "  method            from  \n",
      "  as.sparse.H5Group Seurat\n",
      "\n",
      "── \u001b[1mAttaching packages\u001b[22m ───────────────────────────────────────────────── tidyverse 1.3.1 ──\n",
      "\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.3.3     \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.1.0     \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.8\n",
      "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.1.3     \u001b[32m✔\u001b[39m \u001b[34mstringr\u001b[39m 1.4.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 2.0.1     \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.1\n",
      "\n",
      "── \u001b[1mConflicts\u001b[22m ──────────────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mtidyr\u001b[39m::\u001b[32mexpand()\u001b[39m masks \u001b[34mMatrix\u001b[39m::expand()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m    masks \u001b[34mstats\u001b[39m::lag()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mtidyr\u001b[39m::\u001b[32mpack()\u001b[39m   masks \u001b[34mMatrix\u001b[39m::pack()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mtidyr\u001b[39m::\u001b[32munpack()\u001b[39m masks \u001b[34mMatrix\u001b[39m::unpack()\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(infercnv)\n",
    "library(Matrix)\n",
    "library(Seurat)\n",
    "library(SeuratDisk)\n",
    "library(tidyverse)\n",
    "library(stringr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "dfe28798",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“Unknown file type: h5ad”\n",
      "Creating h5Seurat file for version 3.1.5.9900\n",
      "\n",
      "Adding X as data\n",
      "\n",
      "Adding X as counts\n",
      "\n",
      "Adding meta.features from var\n",
      "\n"
     ]
    }
   ],
   "source": [
    "Convert('/data/countMatrix/rawSingletForR15Kcells.h5ad', assay = 'RNA' ,dest = \"h5seurat\", overwrite = TRUE) # Singlet H5ad - change to your own directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "984cd50c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Validating h5Seurat file\n",
      "\n",
      "Initializing RNA with data\n",
      "\n",
      "Adding counts for RNA\n",
      "\n",
      "Adding feature-level metadata for RNA\n",
      "\n",
      "Adding command information\n",
      "\n",
      "Adding cell-level metadata\n",
      "\n",
      "Adding miscellaneous information\n",
      "\n",
      "Adding tool-specific results\n",
      "\n"
     ]
    }
   ],
   "source": [
    "dat <- LoadH5Seurat('/data/countMatrix/rawSingletForR15Kcells.h5seurat')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "8804bd60",
   "metadata": {},
   "outputs": [],
   "source": [
    "# dat = RenameCells(dat, Idents(dat))\n",
    "dat <- SetIdent(dat, value = \"Barcode\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b1de673b",
   "metadata": {},
   "source": [
    "# inferCNV - ALL samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "f3c18e59",
   "metadata": {},
   "outputs": [],
   "source": [
    "sampleMeta = dat[[c('Barcode', 'guides')]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "5e78c1cd",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.table(\n",
    "    sampleMeta, \n",
    "    file = \"/data/annotations.txt\", \n",
    "    sep = \"\\t\", \n",
    "    row.names = FALSE, \n",
    "    col.names = FALSE, \n",
    "    quote = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "c0c55fad",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>V1</th><th scope=col>V2</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>AAACCCAAGCTAATGA-1-0</td><td>CTTCAAGAGCAACAGTGCTG_+</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>AAACCCAAGTATGTAG-1-0</td><td>GACACCTTCTTCCCCAGCCC_+</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>AAACCCACAAGTCATC-1-0</td><td>GACACCTTCTTCCCCAGCCC_+</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>AAACCCACACACACTA-1-0</td><td>TTCTTCCCCAGCCCAGGTAA_+</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>AAACCCACACCCATAA-1-0</td><td>GACACCTTCTTCCCCAGCCC_+</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>AAACCCACACGCACCA-1-0</td><td>GACACCTTCTTCCCCAGCCC_+</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & V1 & V2\\\\\n",
       "  & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & AAACCCAAGCTAATGA-1-0 & CTTCAAGAGCAACAGTGCTG\\_+\\\\\n",
       "\t2 & AAACCCAAGTATGTAG-1-0 & GACACCTTCTTCCCCAGCCC\\_+\\\\\n",
       "\t3 & AAACCCACAAGTCATC-1-0 & GACACCTTCTTCCCCAGCCC\\_+\\\\\n",
       "\t4 & AAACCCACACACACTA-1-0 & TTCTTCCCCAGCCCAGGTAA\\_+\\\\\n",
       "\t5 & AAACCCACACCCATAA-1-0 & GACACCTTCTTCCCCAGCCC\\_+\\\\\n",
       "\t6 & AAACCCACACGCACCA-1-0 & GACACCTTCTTCCCCAGCCC\\_+\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 2\n",
       "\n",
       "| <!--/--> | V1 &lt;chr&gt; | V2 &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| 1 | AAACCCAAGCTAATGA-1-0 | CTTCAAGAGCAACAGTGCTG_+ |\n",
       "| 2 | AAACCCAAGTATGTAG-1-0 | GACACCTTCTTCCCCAGCCC_+ |\n",
       "| 3 | AAACCCACAAGTCATC-1-0 | GACACCTTCTTCCCCAGCCC_+ |\n",
       "| 4 | AAACCCACACACACTA-1-0 | TTCTTCCCCAGCCCAGGTAA_+ |\n",
       "| 5 | AAACCCACACCCATAA-1-0 | GACACCTTCTTCCCCAGCCC_+ |\n",
       "| 6 | AAACCCACACGCACCA-1-0 | GACACCTTCTTCCCCAGCCC_+ |\n",
       "\n"
      ],
      "text/plain": [
       "  V1                   V2                    \n",
       "1 AAACCCAAGCTAATGA-1-0 CTTCAAGAGCAACAGTGCTG_+\n",
       "2 AAACCCAAGTATGTAG-1-0 GACACCTTCTTCCCCAGCCC_+\n",
       "3 AAACCCACAAGTCATC-1-0 GACACCTTCTTCCCCAGCCC_+\n",
       "4 AAACCCACACACACTA-1-0 TTCTTCCCCAGCCCAGGTAA_+\n",
       "5 AAACCCACACCCATAA-1-0 GACACCTTCTTCCCCAGCCC_+\n",
       "6 AAACCCACACGCACCA-1-0 GACACCTTCTTCCCCAGCCC_+"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "annotations = read.table('/data/annotations.txt', sep = '\\t')\n",
    "head(annotations)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "66615316",
   "metadata": {},
   "outputs": [],
   "source": [
    "dat <- SetIdent(dat, value = \"Barcode\")\n",
    "counts = GetAssayData(object = dat, slot = \"counts\")\n",
    "colnames(counts) <- Idents(object = dat)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "eaff7924",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [2022-02-17 12:29:53] Parsing gene order file: /data/inferCNVgeneName.txt\n",
      "INFO [2022-02-17 12:29:53] Parsing cell annotations file: /data/annotations.txt\n",
      "INFO [2022-02-17 12:29:53] ::order_reduce:Start.\n",
      "INFO [2022-02-17 12:29:55] .order_reduce(): expr and order match.\n",
      "INFO [2022-02-17 12:29:56] ::process_data:order_reduce:Reduction from positional data, new dimensions (r,c) = 36601,33461 Total=356413122 Min=0 Max=2617.\n",
      "INFO [2022-02-17 12:29:57] num genes removed taking into account provided gene ordering list: 1280 = 3.49717220840961% removed.\n",
      "INFO [2022-02-17 12:29:57] -filtering out cells < 100 or > Inf, removing 0 % of cells\n",
      "INFO [2022-02-17 12:30:05] validating infercnv_obj\n"
     ]
    }
   ],
   "source": [
    "infercnv_obj <- CreateInfercnvObject(\n",
    "raw_counts_matrix=counts,\n",
    "annotations_file='/data/annotations.txt',\n",
    "delim=\"\\t\",\n",
    "gene_order_file='/data/inferCNVgeneName.txt',\n",
    "ref_group_names = c('NT')\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "b7293f5e",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [2022-02-17 12:30:13] ::process_data:Start\n",
      "INFO [2022-02-17 12:30:13] Creating output path /data/inferCNVResults15Kcells/\n",
      "INFO [2022-02-17 12:30:13] Checking for saved results.\n",
      "INFO [2022-02-17 12:30:13] \n",
      "\n",
      "\tSTEP 1: incoming data\n",
      "\n",
      "INFO [2022-02-17 12:31:52] \n",
      "\n",
      "\tSTEP 02: Removing lowly expressed genes\n",
      "\n",
      "INFO [2022-02-17 12:31:52] ::above_min_mean_expr_cutoff:Start\n",
      "INFO [2022-02-17 12:31:56] Removing 28149 genes from matrix as below mean expr threshold: 0.1\n",
      "INFO [2022-02-17 12:32:00] validating infercnv_obj\n",
      "INFO [2022-02-17 12:32:00] There are 7172 genes and 33461 cells remaining in the expr matrix.\n",
      "INFO [2022-02-17 12:32:17] no genes removed due to min cells/gene filter\n",
      "INFO [2022-02-17 12:33:58] \n",
      "\n",
      "\tSTEP 03: normalization by sequencing depth\n",
      "\n",
      "INFO [2022-02-17 12:33:58] normalizing counts matrix by depth\n",
      "INFO [2022-02-17 12:34:20] Computed total sum normalization factor as median libsize: 7642.000000\n",
      "INFO [2022-02-17 12:34:20] Adding h-spike\n",
      "INFO [2022-02-17 12:34:20] -hspike modeling of NT\n",
      "INFO [2022-02-17 12:36:23] validating infercnv_obj\n",
      "INFO [2022-02-17 12:36:23] normalizing counts matrix by depth\n",
      "INFO [2022-02-17 12:36:23] Using specified normalization factor: 7642.000000\n",
      "INFO [2022-02-17 12:38:00] \n",
      "\n",
      "\tSTEP 04: log transformation of data\n",
      "\n",
      "INFO [2022-02-17 12:38:00] transforming log2xplus1()\n",
      "INFO [2022-02-17 12:38:13] -mirroring for hspike\n",
      "INFO [2022-02-17 12:38:13] transforming log2xplus1()\n",
      "INFO [2022-02-17 12:39:45] \n",
      "\n",
      "\tSTEP 08: removing average of reference data (before smoothing)\n",
      "\n",
      "INFO [2022-02-17 12:39:45] ::subtract_ref_expr_from_obs:Start inv_log=FALSE, use_bounds=TRUE\n",
      "INFO [2022-02-17 12:39:45] subtracting mean(normal) per gene per cell across all data\n",
      "INFO [2022-02-17 12:39:56] -subtracting expr per gene, use_bounds=TRUE\n",
      "INFO [2022-02-17 14:33:38] -mirroring for hspike\n",
      "INFO [2022-02-17 14:33:38] ::subtract_ref_expr_from_obs:Start inv_log=FALSE, use_bounds=TRUE\n",
      "INFO [2022-02-17 14:33:38] subtracting mean(normal) per gene per cell across all data\n",
      "INFO [2022-02-17 14:33:40] -subtracting expr per gene, use_bounds=TRUE\n",
      "INFO [2022-02-17 14:36:08] \n",
      "\n",
      "\tSTEP 09: apply max centered expression threshold: 3\n",
      "\n",
      "INFO [2022-02-17 14:36:09] ::process_data:setting max centered expr, threshold set to: +/-:  3\n",
      "INFO [2022-02-17 14:36:16] -mirroring for hspike\n",
      "INFO [2022-02-17 14:36:16] ::process_data:setting max centered expr, threshold set to: +/-:  3\n",
      "INFO [2022-02-17 14:38:44] \n",
      "\n",
      "\tSTEP 10: Smoothing data per cell by chromosome\n",
      "\n",
      "INFO [2022-02-17 14:38:44] smooth_by_chromosome: chr: chr1\n",
      "INFO [2022-02-17 14:39:24] smooth_by_chromosome: chr: chr2\n",
      "INFO [2022-02-17 14:40:00] smooth_by_chromosome: chr: chr3\n",
      "INFO [2022-02-17 14:40:33] smooth_by_chromosome: chr: chr4\n",
      "INFO [2022-02-17 14:41:03] smooth_by_chromosome: chr: chr5\n",
      "INFO [2022-02-17 14:41:35] smooth_by_chromosome: chr: chr6\n",
      "INFO [2022-02-17 14:42:09] smooth_by_chromosome: chr: chr7\n",
      "INFO [2022-02-17 14:42:40] smooth_by_chromosome: chr: chr8\n",
      "INFO [2022-02-17 14:43:09] smooth_by_chromosome: chr: chr9\n",
      "INFO [2022-02-17 14:43:39] smooth_by_chromosome: chr: chr10\n",
      "INFO [2022-02-17 14:44:09] smooth_by_chromosome: chr: chr11\n",
      "INFO [2022-02-17 14:44:43] smooth_by_chromosome: chr: chr12\n",
      "INFO [2022-02-17 14:45:16] smooth_by_chromosome: chr: chr13\n",
      "INFO [2022-02-17 14:45:43] smooth_by_chromosome: chr: chr14\n",
      "INFO [2022-02-17 14:46:13] smooth_by_chromosome: chr: chr15\n",
      "INFO [2022-02-17 14:46:41] smooth_by_chromosome: chr: chr16\n",
      "INFO [2022-02-17 14:47:13] smooth_by_chromosome: chr: chr17\n",
      "INFO [2022-02-17 14:47:47] smooth_by_chromosome: chr: chr18\n",
      "INFO [2022-02-17 14:48:13] smooth_by_chromosome: chr: chr19\n",
      "INFO [2022-02-17 14:48:48] smooth_by_chromosome: chr: chr20\n",
      "INFO [2022-02-17 14:49:15] smooth_by_chromosome: chr: chr21\n",
      "INFO [2022-02-17 14:49:32] smooth_by_chromosome: chr: chr22\n",
      "INFO [2022-02-17 14:50:00] -mirroring for hspike\n",
      "INFO [2022-02-17 14:50:00] smooth_by_chromosome: chr: chrA\n",
      "INFO [2022-02-17 14:50:00] smooth_by_chromosome: chr: chr_0\n",
      "INFO [2022-02-17 14:50:00] smooth_by_chromosome: chr: chr_B\n",
      "INFO [2022-02-17 14:50:00] smooth_by_chromosome: chr: chr_0pt5\n",
      "INFO [2022-02-17 14:50:00] smooth_by_chromosome: chr: chr_C\n",
      "INFO [2022-02-17 14:50:01] smooth_by_chromosome: chr: chr_1pt5\n",
      "INFO [2022-02-17 14:50:01] smooth_by_chromosome: chr: chr_D\n",
      "INFO [2022-02-17 14:50:01] smooth_by_chromosome: chr: chr_2pt0\n",
      "INFO [2022-02-17 14:50:02] smooth_by_chromosome: chr: chr_E\n",
      "INFO [2022-02-17 14:50:02] smooth_by_chromosome: chr: chr_3pt0\n",
      "INFO [2022-02-17 14:50:02] smooth_by_chromosome: chr: chr_F\n",
      "INFO [2022-02-17 14:52:29] \n",
      "\n",
      "\tSTEP 11: re-centering data across chromosome after smoothing\n",
      "\n",
      "INFO [2022-02-17 14:52:29] ::center_smooth across chromosomes per cell\n",
      "INFO [2022-02-17 14:53:14] -mirroring for hspike\n",
      "INFO [2022-02-17 14:53:14] ::center_smooth across chromosomes per cell\n",
      "INFO [2022-02-17 14:55:45] \n",
      "\n",
      "\tSTEP 12: removing average of reference data (after smoothing)\n",
      "\n",
      "INFO [2022-02-17 14:55:45] ::subtract_ref_expr_from_obs:Start inv_log=FALSE, use_bounds=TRUE\n",
      "INFO [2022-02-17 14:55:45] subtracting mean(normal) per gene per cell across all data\n",
      "INFO [2022-02-17 14:55:55] -subtracting expr per gene, use_bounds=TRUE\n",
      "INFO [2022-02-17 14:56:15] -mirroring for hspike\n",
      "INFO [2022-02-17 14:56:15] ::subtract_ref_expr_from_obs:Start inv_log=FALSE, use_bounds=TRUE\n",
      "INFO [2022-02-17 14:56:15] subtracting mean(normal) per gene per cell across all data\n",
      "INFO [2022-02-17 14:56:17] -subtracting expr per gene, use_bounds=TRUE\n",
      "INFO [2022-02-17 14:58:43] \n",
      "\n",
      "\tSTEP 14: invert log2(FC) to FC\n",
      "\n",
      "INFO [2022-02-17 14:58:43] invert_log2(), computing 2^x\n",
      "INFO [2022-02-17 14:58:58] -mirroring for hspike\n",
      "INFO [2022-02-17 14:58:58] invert_log2(), computing 2^x\n",
      "INFO [2022-02-17 15:02:18] \n",
      "\n",
      "\tSTEP 15: Clustering samples (not defining tumor subclusters)\n",
      "\n",
      "INFO [2022-02-17 15:02:18] define_signif_tumor_subclusters(p_val=0.1\n",
      "INFO [2022-02-17 15:02:18] define_signif_tumor_subclusters(), tumor: CTTCAAGAGCAACAGTGCTG_+\n",
      "INFO [2022-02-17 15:20:37] cut tree into: 1 groups\n",
      "INFO [2022-02-17 15:20:37] -processing CTTCAAGAGCAACAGTGCTG_+,CTTCAAGAGCAACAGTGCTG_+_s1\n",
      "INFO [2022-02-17 15:20:37] define_signif_tumor_subclusters(), tumor: GACACCTTCTTCCCCAGCCC_+\n",
      "INFO [2022-02-17 15:31:17] cut tree into: 1 groups\n",
      "INFO [2022-02-17 15:31:17] -processing GACACCTTCTTCCCCAGCCC_+,GACACCTTCTTCCCCAGCCC_+_s1\n",
      "INFO [2022-02-17 15:31:17] define_signif_tumor_subclusters(), tumor: TTCTTCCCCAGCCCAGGTAA_+\n",
      "INFO [2022-02-17 15:43:34] cut tree into: 1 groups\n",
      "INFO [2022-02-17 15:43:34] -processing TTCTTCCCCAGCCCAGGTAA_+,TTCTTCCCCAGCCCAGGTAA_+_s1\n",
      "INFO [2022-02-17 15:43:34] define_signif_tumor_subclusters(), tumor: AACAAATGTGTCACAAAGTA_+\n",
      "INFO [2022-02-17 15:52:24] cut tree into: 1 groups\n",
      "INFO [2022-02-17 15:52:24] -processing AACAAATGTGTCACAAAGTA_+,AACAAATGTGTCACAAAGTA_+_s1\n",
      "INFO [2022-02-17 15:52:24] define_signif_tumor_subclusters(), tumor: TCTCTCAGCTGGTACACGGC_-\n",
      "INFO [2022-02-17 16:09:25] cut tree into: 1 groups\n",
      "INFO [2022-02-17 16:09:25] -processing TCTCTCAGCTGGTACACGGC_-,TCTCTCAGCTGGTACACGGC_-_s1\n",
      "INFO [2022-02-17 16:09:25] define_signif_tumor_subclusters(), tumor: GCTGGTACACGGCAGGGTCA_-\n",
      "INFO [2022-02-17 16:21:39] cut tree into: 1 groups\n",
      "INFO [2022-02-17 16:21:39] -processing GCTGGTACACGGCAGGGTCA_-,GCTGGTACACGGCAGGGTCA_-_s1\n",
      "INFO [2022-02-17 16:21:39] define_signif_tumor_subclusters(), tumor: GAGAATCAAAATCGGTGAAT_-\n",
      "INFO [2022-02-17 16:32:25] cut tree into: 1 groups\n",
      "INFO [2022-02-17 16:32:25] -processing GAGAATCAAAATCGGTGAAT_-,GAGAATCAAAATCGGTGAAT_-_s1\n",
      "INFO [2022-02-17 16:32:25] define_signif_tumor_subclusters(), tumor: CTTCTTCCCCAGCCCAGGTA_+\n",
      "INFO [2022-02-17 16:35:32] cut tree into: 1 groups\n",
      "INFO [2022-02-17 16:35:32] -processing CTTCTTCCCCAGCCCAGGTA_+,CTTCTTCCCCAGCCCAGGTA_+_s1\n",
      "INFO [2022-02-17 16:35:32] define_signif_tumor_subclusters(), tumor: CTCTCAGCTGGTACACGGCA_-\n",
      "INFO [2022-02-17 16:53:02] cut tree into: 1 groups\n",
      "INFO [2022-02-17 16:53:02] -processing CTCTCAGCTGGTACACGGCA_-,CTCTCAGCTGGTACACGGCA_-_s1\n",
      "INFO [2022-02-17 16:53:02] define_signif_tumor_subclusters(), tumor: CTRL_June\n",
      "INFO [2022-02-17 17:05:32] cut tree into: 1 groups\n",
      "INFO [2022-02-17 17:05:32] -processing CTRL_June,CTRL_June_s1\n",
      "INFO [2022-02-17 17:05:32] define_signif_tumor_subclusters(), tumor: CTTACCTGGGCTGGGGAAGA_-\n",
      "INFO [2022-02-17 17:19:22] cut tree into: 1 groups\n",
      "INFO [2022-02-17 17:19:22] -processing CTTACCTGGGCTGGGGAAGA_-,CTTACCTGGGCTGGGGAAGA_-_s1\n",
      "INFO [2022-02-17 17:19:22] define_signif_tumor_subclusters(), tumor: NT\n",
      "INFO [2022-02-17 17:29:46] cut tree into: 1 groups\n",
      "INFO [2022-02-17 17:29:46] -processing NT,NT_s1\n",
      "INFO [2022-02-17 17:29:46] -mirroring for hspike\n",
      "INFO [2022-02-17 17:29:46] define_signif_tumor_subclusters(p_val=0.1\n",
      "INFO [2022-02-17 17:29:46] define_signif_tumor_subclusters(), tumor: spike_tumor_cell_NT\n",
      "INFO [2022-02-17 17:29:47] cut tree into: 1 groups\n",
      "INFO [2022-02-17 17:29:47] -processing spike_tumor_cell_NT,spike_tumor_cell_NT_s1\n",
      "INFO [2022-02-17 17:29:47] define_signif_tumor_subclusters(), tumor: simnorm_cell_NT\n",
      "INFO [2022-02-17 17:29:47] cut tree into: 1 groups\n",
      "INFO [2022-02-17 17:29:47] -processing simnorm_cell_NT,simnorm_cell_NT_s1\n",
      "INFO [2022-02-17 17:36:26] ::plot_cnv:Start\n",
      "INFO [2022-02-17 17:36:26] ::plot_cnv:Current data dimensions (r,c)=7172,33461 Total=240292575.758159 Min=0.62671765031926 Max=3.4839070923135.\n",
      "INFO [2022-02-17 17:36:29] ::plot_cnv:Depending on the size of the matrix this may take a moment.\n",
      "INFO [2022-02-17 17:40:58] plot_cnv(): auto thresholding at: (0.858358 , 1.144228)\n",
      "INFO [2022-02-17 17:41:09] plot_cnv_observation:Start\n",
      "INFO [2022-02-17 17:41:09] Observation data size: Cells= 30863 Genes= 7172\n",
      "INFO [2022-02-17 17:41:15] plot_cnv_observation:Writing observation groupings/color.\n",
      "INFO [2022-02-17 17:41:15] plot_cnv_observation:Done writing observation groupings/color.\n",
      "INFO [2022-02-17 17:41:16] plot_cnv_observation:Writing observation heatmap thresholds.\n",
      "INFO [2022-02-17 17:41:16] plot_cnv_observation:Done writing observation heatmap thresholds.\n",
      "INFO [2022-02-17 17:42:20] Colors for breaks:  #00008B,#24249B,#4848AB,#6D6DBC,#9191CC,#B6B6DD,#DADAEE,#FFFFFF,#EEDADA,#DDB6B6,#CC9191,#BC6D6D,#AB4848,#9B2424,#8B0000\n",
      "INFO [2022-02-17 17:42:20] Quantiles of plotted data range: 0.858357819183466,0.968535076763235,0.998740928384752,1.03087631627448,1.14422806959669\n",
      "INFO [2022-02-17 17:43:03] plot_cnv_observations:Writing observation data to /data/inferCNVResults15Kcells//infercnv.preliminary.observations.txt\n",
      "INFO [2022-02-17 17:47:03] plot_cnv_references:Start\n",
      "INFO [2022-02-17 17:47:03] Reference data size: Cells= 2598 Genes= 7172\n",
      "INFO [2022-02-17 17:57:22] plot_cnv_references:Number reference groups= 1\n",
      "INFO [2022-02-17 17:57:22] plot_cnv_references:Plotting heatmap.\n",
      "INFO [2022-02-17 17:57:26] Colors for breaks:  #00008B,#24249B,#4848AB,#6D6DBC,#9191CC,#B6B6DD,#DADAEE,#FFFFFF,#EEDADA,#DDB6B6,#CC9191,#BC6D6D,#AB4848,#9B2424,#8B0000\n",
      "INFO [2022-02-17 17:57:26] Quantiles of plotted data range: 0.858357819183466,0.969725598044743,0.998741092939259,1.02967743344773,1.14422806959669\n",
      "INFO [2022-02-17 17:57:28] plot_cnv_references:Writing reference data to /data/inferCNVResults15Kcells//infercnv.preliminary.references.txt\n",
      "INFO [2022-02-17 17:57:48] \n",
      "\n",
      "\tSTEP 17: HMM-based CNV prediction\n",
      "\n",
      "INFO [2022-02-17 17:57:48] predict_CNV_via_HMM_on_whole_tumor_samples\n",
      "INFO [2022-02-17 17:57:56] -done predicting CNV based on initial tumor subclusters\n",
      "INFO [2022-02-17 17:59:03] get_predicted_CNV_regions(subcluster)\n",
      "INFO [2022-02-17 17:59:03] -processing cell_group_name: CTTCAAGAGCAACAGTGCTG_+.CTTCAAGAGCAACAGTGCTG_+_s1, size: 3439\n",
      "INFO [2022-02-17 17:59:28] -processing cell_group_name: GACACCTTCTTCCCCAGCCC_+.GACACCTTCTTCCCCAGCCC_+_s1, size: 2679\n",
      "INFO [2022-02-17 17:59:49] -processing cell_group_name: TTCTTCCCCAGCCCAGGTAA_+.TTCTTCCCCAGCCCAGGTAA_+_s1, size: 2787\n",
      "INFO [2022-02-17 18:00:09] -processing cell_group_name: AACAAATGTGTCACAAAGTA_+.AACAAATGTGTCACAAAGTA_+_s1, size: 2431\n",
      "INFO [2022-02-17 18:00:28] -processing cell_group_name: TCTCTCAGCTGGTACACGGC_-.TCTCTCAGCTGGTACACGGC_-_s1, size: 3278\n",
      "INFO [2022-02-17 18:00:52] -processing cell_group_name: GCTGGTACACGGCAGGGTCA_-.GCTGGTACACGGCAGGGTCA_-_s1, size: 2861\n",
      "INFO [2022-02-17 18:01:13] -processing cell_group_name: GAGAATCAAAATCGGTGAAT_-.GAGAATCAAAATCGGTGAAT_-_s1, size: 2685\n",
      "INFO [2022-02-17 18:01:33] -processing cell_group_name: CTTCTTCCCCAGCCCAGGTA_+.CTTCTTCCCCAGCCCAGGTA_+_s1, size: 1466\n",
      "INFO [2022-02-17 18:01:47] -processing cell_group_name: CTCTCAGCTGGTACACGGCA_-.CTCTCAGCTGGTACACGGCA_-_s1, size: 3359\n",
      "INFO [2022-02-17 18:02:11] -processing cell_group_name: CTRL_June.CTRL_June_s1, size: 2845\n",
      "INFO [2022-02-17 18:02:32] -processing cell_group_name: CTTACCTGGGCTGGGGAAGA_-.CTTACCTGGGCTGGGGAAGA_-_s1, size: 3033\n",
      "INFO [2022-02-17 18:02:54] -processing cell_group_name: NT.NT_s1, size: 2598\n",
      "INFO [2022-02-17 18:03:14] -writing cell clusters file: /data/inferCNVResults15Kcells//17_HMM_predHMMi6.hmm_mode-samples.cell_groupings\n",
      "INFO [2022-02-17 18:03:14] -writing cnv regions file: /data/inferCNVResults15Kcells//17_HMM_predHMMi6.hmm_mode-samples.pred_cnv_regions.dat\n",
      "INFO [2022-02-17 18:03:14] -writing per-gene cnv report: /data/inferCNVResults15Kcells//17_HMM_predHMMi6.hmm_mode-samples.pred_cnv_genes.dat\n",
      "INFO [2022-02-17 18:03:14] -writing gene ordering info: /data/inferCNVResults15Kcells//17_HMM_predHMMi6.hmm_mode-samples.genes_used.dat\n",
      "INFO [2022-02-17 18:03:15] ::plot_cnv:Start\n",
      "INFO [2022-02-17 18:03:15] ::plot_cnv:Current data dimensions (r,c)=7172,33461 Total=719946876 Min=3 Max=3.\n",
      "INFO [2022-02-17 18:03:17] ::plot_cnv:Depending on the size of the matrix this may take a moment.\n",
      "INFO [2022-02-17 18:07:05] plot_cnv_observation:Start\n",
      "INFO [2022-02-17 18:07:05] Observation data size: Cells= 30863 Genes= 7172\n",
      "INFO [2022-02-17 18:07:11] plot_cnv_observation:Writing observation groupings/color.\n",
      "INFO [2022-02-17 18:07:11] plot_cnv_observation:Done writing observation groupings/color.\n",
      "INFO [2022-02-17 18:07:12] plot_cnv_observation:Writing observation heatmap thresholds.\n",
      "INFO [2022-02-17 18:07:12] plot_cnv_observation:Done writing observation heatmap thresholds.\n",
      "INFO [2022-02-17 18:08:17] Colors for breaks:  #00008B,#24249B,#4848AB,#6D6DBC,#9191CC,#B6B6DD,#DADAEE,#FFFFFF,#EEDADA,#DDB6B6,#CC9191,#BC6D6D,#AB4848,#9B2424,#8B0000\n",
      "INFO [2022-02-17 18:08:17] Quantiles of plotted data range: 3,3,3,3,3\n",
      "INFO [2022-02-17 18:08:58] plot_cnv_observations:Writing observation data to /data/inferCNVResults15Kcells//infercnv.17_HMM_predHMMi6.hmm_mode-samples.observations.txt\n",
      "INFO [2022-02-17 18:12:25] plot_cnv_references:Start\n",
      "INFO [2022-02-17 18:12:25] Reference data size: Cells= 2598 Genes= 7172\n",
      "INFO [2022-02-17 18:22:52] plot_cnv_references:Number reference groups= 1\n",
      "INFO [2022-02-17 18:22:52] plot_cnv_references:Plotting heatmap.\n",
      "INFO [2022-02-17 18:22:57] Colors for breaks:  #00008B,#24249B,#4848AB,#6D6DBC,#9191CC,#B6B6DD,#DADAEE,#FFFFFF,#EEDADA,#DDB6B6,#CC9191,#BC6D6D,#AB4848,#9B2424,#8B0000\n",
      "INFO [2022-02-17 18:22:57] Quantiles of plotted data range: 3,3,3,3,3\n",
      "INFO [2022-02-17 18:22:59] plot_cnv_references:Writing reference data to /data/inferCNVResults15Kcells//infercnv.17_HMM_predHMMi6.hmm_mode-samples.references.txt\n",
      "INFO [2022-02-17 18:23:28] \n",
      "\n",
      "\tSTEP 19: Converting HMM-based CNV states to repr expr vals\n",
      "\n",
      "INFO [2022-02-17 18:24:50] ::plot_cnv:Start\n",
      "INFO [2022-02-17 18:24:50] ::plot_cnv:Current data dimensions (r,c)=7172,33461 Total=239982292 Min=1 Max=1.\n",
      "INFO [2022-02-17 18:24:52] ::plot_cnv:Depending on the size of the matrix this may take a moment.\n",
      "INFO [2022-02-17 18:28:34] plot_cnv_observation:Start\n",
      "INFO [2022-02-17 18:28:34] Observation data size: Cells= 30863 Genes= 7172\n",
      "INFO [2022-02-17 18:28:39] plot_cnv_observation:Writing observation groupings/color.\n",
      "INFO [2022-02-17 18:28:40] plot_cnv_observation:Done writing observation groupings/color.\n",
      "INFO [2022-02-17 18:28:41] plot_cnv_observation:Writing observation heatmap thresholds.\n",
      "INFO [2022-02-17 18:28:41] plot_cnv_observation:Done writing observation heatmap thresholds.\n",
      "INFO [2022-02-17 18:29:43] Colors for breaks:  #00008B,#24249B,#4848AB,#6D6DBC,#9191CC,#B6B6DD,#DADAEE,#FFFFFF,#EEDADA,#DDB6B6,#CC9191,#BC6D6D,#AB4848,#9B2424,#8B0000\n",
      "INFO [2022-02-17 18:29:43] Quantiles of plotted data range: 1,1,1,1,1\n",
      "INFO [2022-02-17 18:30:23] plot_cnv_observations:Writing observation data to /data/inferCNVResults15Kcells//infercnv.19_HMM_predHMMi6.hmm_mode-samples.Pnorm_0.5.repr_intensities.observations.txt\n",
      "INFO [2022-02-17 18:33:44] plot_cnv_references:Start\n",
      "INFO [2022-02-17 18:33:44] Reference data size: Cells= 2598 Genes= 7172\n",
      "INFO [2022-02-17 18:44:06] plot_cnv_references:Number reference groups= 1\n",
      "INFO [2022-02-17 18:44:06] plot_cnv_references:Plotting heatmap.\n",
      "INFO [2022-02-17 18:44:10] Colors for breaks:  #00008B,#24249B,#4848AB,#6D6DBC,#9191CC,#B6B6DD,#DADAEE,#FFFFFF,#EEDADA,#DDB6B6,#CC9191,#BC6D6D,#AB4848,#9B2424,#8B0000\n",
      "INFO [2022-02-17 18:44:10] Quantiles of plotted data range: 1,1,1,1,1\n",
      "INFO [2022-02-17 18:44:12] plot_cnv_references:Writing reference data to /data/inferCNVResults15Kcells//infercnv.19_HMM_predHMMi6.hmm_mode-samples.Pnorm_0.5.repr_intensities.references.txt\n",
      "INFO [2022-02-17 18:44:29] \n",
      "\n",
      "\tSTEP 21: Denoising\n",
      "\n",
      "INFO [2022-02-17 18:44:29] ::process_data:Remove noise, noise threshold defined via ref mean sd_amplifier:  1.5\n",
      "INFO [2022-02-17 18:44:29] denoising using mean(normal) +- sd_amplifier * sd(normal) per gene per cell across all data\n",
      "INFO [2022-02-17 18:44:29] :: **** clear_noise_via_ref_quantiles **** : removing noise between bounds:  0.92956168081626 - 1.0728384449233\n",
      "INFO [2022-02-17 18:45:55] ::plot_cnv:Start\n",
      "INFO [2022-02-17 18:45:55] ::plot_cnv:Current data dimensions (r,c)=7172,33461 Total=240769555.871205 Min=0.62671765031926 Max=3.4839070923135.\n",
      "INFO [2022-02-17 18:45:57] ::plot_cnv:Depending on the size of the matrix this may take a moment.\n",
      "INFO [2022-02-17 18:50:16] plot_cnv(): auto thresholding at: (0.862333 , 1.144228)\n",
      "INFO [2022-02-17 18:50:26] plot_cnv_observation:Start\n",
      "INFO [2022-02-17 18:50:26] Observation data size: Cells= 30863 Genes= 7172\n",
      "INFO [2022-02-17 18:50:31] plot_cnv_observation:Writing observation groupings/color.\n",
      "INFO [2022-02-17 18:50:31] plot_cnv_observation:Done writing observation groupings/color.\n",
      "INFO [2022-02-17 18:50:33] plot_cnv_observation:Writing observation heatmap thresholds.\n",
      "INFO [2022-02-17 18:50:33] plot_cnv_observation:Done writing observation heatmap thresholds.\n",
      "INFO [2022-02-17 18:51:35] Colors for breaks:  #00008B,#24249B,#4848AB,#6D6DBC,#9191CC,#B6B6DD,#DADAEE,#FFFFFF,#EEDADA,#DDB6B6,#CC9191,#BC6D6D,#AB4848,#9B2424,#8B0000\n",
      "INFO [2022-02-17 18:51:35] Quantiles of plotted data range: 0.862332946761686,1.00120006286978,1.00120006286978,1.00120006286978,1.14422806959669\n",
      "INFO [2022-02-17 18:52:16] plot_cnv_observations:Writing observation data to /data/inferCNVResults15Kcells//infercnv.21_denoised.observations.txt\n",
      "INFO [2022-02-17 18:56:05] plot_cnv_references:Start\n",
      "INFO [2022-02-17 18:56:05] Reference data size: Cells= 2598 Genes= 7172\n",
      "INFO [2022-02-17 19:06:21] plot_cnv_references:Number reference groups= 1\n",
      "INFO [2022-02-17 19:06:21] plot_cnv_references:Plotting heatmap.\n",
      "INFO [2022-02-17 19:06:25] Colors for breaks:  #00008B,#24249B,#4848AB,#6D6DBC,#9191CC,#B6B6DD,#DADAEE,#FFFFFF,#EEDADA,#DDB6B6,#CC9191,#BC6D6D,#AB4848,#9B2424,#8B0000\n",
      "INFO [2022-02-17 19:06:25] Quantiles of plotted data range: 0.862332946761686,1.00120006286978,1.00120006286978,1.00120006286978,1.14422806959669\n",
      "INFO [2022-02-17 19:06:27] plot_cnv_references:Writing reference data to /data/inferCNVResults15Kcells//infercnv.21_denoised.references.txt\n",
      "INFO [2022-02-17 19:08:02] \n",
      "\n",
      "## Making the final infercnv heatmap ##\n",
      "INFO [2022-02-17 19:08:03] ::plot_cnv:Start\n",
      "INFO [2022-02-17 19:08:03] ::plot_cnv:Current data dimensions (r,c)=7172,33461 Total=240769555.871205 Min=0.62671765031926 Max=3.4839070923135.\n",
      "INFO [2022-02-17 19:08:06] ::plot_cnv:Depending on the size of the matrix this may take a moment.\n",
      "INFO [2022-02-17 19:12:25] plot_cnv(): auto thresholding at: (0.855772 , 1.144228)\n",
      "INFO [2022-02-17 19:12:35] plot_cnv_observation:Start\n",
      "INFO [2022-02-17 19:12:35] Observation data size: Cells= 30863 Genes= 7172\n",
      "INFO [2022-02-17 19:12:41] plot_cnv_observation:Writing observation groupings/color.\n",
      "INFO [2022-02-17 19:12:41] plot_cnv_observation:Done writing observation groupings/color.\n",
      "INFO [2022-02-17 19:12:41] plot_cnv_observation:Writing observation heatmap thresholds.\n",
      "INFO [2022-02-17 19:12:41] plot_cnv_observation:Done writing observation heatmap thresholds.\n",
      "INFO [2022-02-17 19:13:44] Colors for breaks:  #00008B,#24249B,#4848AB,#6D6DBC,#9191CC,#B6B6DD,#DADAEE,#FFFFFF,#EEDADA,#DDB6B6,#CC9191,#BC6D6D,#AB4848,#9B2424,#8B0000\n",
      "INFO [2022-02-17 19:13:44] Quantiles of plotted data range: 0.85577193040331,1.00120006286978,1.00120006286978,1.00120006286978,1.14422806959669\n",
      "INFO [2022-02-17 19:14:25] plot_cnv_observations:Writing observation data to /data/inferCNVResults15Kcells//infercnv.observations.txt\n",
      "INFO [2022-02-17 19:18:15] plot_cnv_references:Start\n",
      "INFO [2022-02-17 19:18:15] Reference data size: Cells= 2598 Genes= 7172\n",
      "INFO [2022-02-17 19:28:35] plot_cnv_references:Number reference groups= 1\n",
      "INFO [2022-02-17 19:28:35] plot_cnv_references:Plotting heatmap.\n",
      "INFO [2022-02-17 19:28:39] Colors for breaks:  #00008B,#24249B,#4848AB,#6D6DBC,#9191CC,#B6B6DD,#DADAEE,#FFFFFF,#EEDADA,#DDB6B6,#CC9191,#BC6D6D,#AB4848,#9B2424,#8B0000\n",
      "INFO [2022-02-17 19:28:39] Quantiles of plotted data range: 0.85577193040331,1.00120006286978,1.00120006286978,1.00120006286978,1.14422806959669\n",
      "INFO [2022-02-17 19:28:41] plot_cnv_references:Writing reference data to /data/inferCNVResults15Kcells//infercnv.references.txt\n"
     ]
    }
   ],
   "source": [
    "infercnv_obj = infercnv::run(\n",
    "    infercnv_obj,\n",
    "    cutoff=0.1, # cutoff=1 works well for Smart-seq2, and cutoff=0.1 works well for 10x Genomics\n",
    "    out_dir='/data/inferCNVResults15Kcells/', \n",
    "    cluster_by_groups=TRUE, \n",
    "    denoise=TRUE,\n",
    "    HMM=TRUE,\n",
    "    num_threads=24,\n",
    ")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "c2b0a515",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 6 × 33461 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>AAACCCAAGCTAATGA-1-0</th><th scope=col>AAACCCAAGTATGTAG-1-0</th><th scope=col>AAACCCACAAGTCATC-1-0</th><th scope=col>AAACCCACACACACTA-1-0</th><th scope=col>AAACCCACACCCATAA-1-0</th><th scope=col>AAACCCACACGCACCA-1-0</th><th scope=col>AAACCCACACTGCGAC-1-0</th><th scope=col>AAACCCACAGGACTTT-1-0</th><th scope=col>AAACCCAGTTGGGTTT-1-0</th><th scope=col>AAACCCATCAGGTAAA-1-0</th><th scope=col>⋯</th><th scope=col>TTTGTTGCAATCCTTT-1-2</th><th scope=col>TTTGTTGCATGACTAC-1-2</th><th scope=col>TTTGTTGGTCAAGTTC-1-2</th><th scope=col>TTTGTTGGTGGATTTC-1-2</th><th scope=col>TTTGTTGGTTACAGCT-1-2</th><th scope=col>TTTGTTGGTTCTTGTT-1-2</th><th scope=col>TTTGTTGGTTGCCGAC-1-2</th><th scope=col>TTTGTTGTCCGGTAGC-1-2</th><th scope=col>TTTGTTGTCTAGAACC-1-2</th><th scope=col>TTTGTTGTCTTCTGGC-1-2</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>NOC2L</th><td>1.0012</td><td>0.8496507</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>⋯</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>0.8995852</td><td>0.8588233</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td></tr>\n",
       "\t<tr><th scope=row>ISG15</th><td>1.0012</td><td>0.8509542</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>⋯</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>0.8996055</td><td>0.8595151</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td></tr>\n",
       "\t<tr><th scope=row>TNFRSF4</th><td>1.0012</td><td>0.8522806</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>⋯</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>0.8997827</td><td>0.8602901</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td></tr>\n",
       "\t<tr><th scope=row>SDF4</th><td>1.0012</td><td>0.8534545</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>⋯</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>0.8999158</td><td>0.8609641</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td></tr>\n",
       "\t<tr><th scope=row>B3GALT6</th><td>1.0012</td><td>0.8546630</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>⋯</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>0.9001891</td><td>0.8617201</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td></tr>\n",
       "\t<tr><th scope=row>UBE2J2</th><td>1.0012</td><td>0.8557378</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>⋯</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>0.9004141</td><td>0.8623833</td><td>1.0012</td><td>1.0012</td><td>1.0012</td><td>1.0012</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 6 × 33461 of type dbl\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & AAACCCAAGCTAATGA-1-0 & AAACCCAAGTATGTAG-1-0 & AAACCCACAAGTCATC-1-0 & AAACCCACACACACTA-1-0 & AAACCCACACCCATAA-1-0 & AAACCCACACGCACCA-1-0 & AAACCCACACTGCGAC-1-0 & AAACCCACAGGACTTT-1-0 & AAACCCAGTTGGGTTT-1-0 & AAACCCATCAGGTAAA-1-0 & ⋯ & TTTGTTGCAATCCTTT-1-2 & TTTGTTGCATGACTAC-1-2 & TTTGTTGGTCAAGTTC-1-2 & TTTGTTGGTGGATTTC-1-2 & TTTGTTGGTTACAGCT-1-2 & TTTGTTGGTTCTTGTT-1-2 & TTTGTTGGTTGCCGAC-1-2 & TTTGTTGTCCGGTAGC-1-2 & TTTGTTGTCTAGAACC-1-2 & TTTGTTGTCTTCTGGC-1-2\\\\\n",
       "\\hline\n",
       "\tNOC2L & 1.0012 & 0.8496507 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & ⋯ & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 0.8995852 & 0.8588233 & 1.0012 & 1.0012 & 1.0012 & 1.0012\\\\\n",
       "\tISG15 & 1.0012 & 0.8509542 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & ⋯ & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 0.8996055 & 0.8595151 & 1.0012 & 1.0012 & 1.0012 & 1.0012\\\\\n",
       "\tTNFRSF4 & 1.0012 & 0.8522806 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & ⋯ & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 0.8997827 & 0.8602901 & 1.0012 & 1.0012 & 1.0012 & 1.0012\\\\\n",
       "\tSDF4 & 1.0012 & 0.8534545 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & ⋯ & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 0.8999158 & 0.8609641 & 1.0012 & 1.0012 & 1.0012 & 1.0012\\\\\n",
       "\tB3GALT6 & 1.0012 & 0.8546630 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & ⋯ & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 0.9001891 & 0.8617201 & 1.0012 & 1.0012 & 1.0012 & 1.0012\\\\\n",
       "\tUBE2J2 & 1.0012 & 0.8557378 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 1.0012 & ⋯ & 1.0012 & 1.0012 & 1.0012 & 1.0012 & 0.9004141 & 0.8623833 & 1.0012 & 1.0012 & 1.0012 & 1.0012\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 6 × 33461 of type dbl\n",
       "\n",
       "| <!--/--> | AAACCCAAGCTAATGA-1-0 | AAACCCAAGTATGTAG-1-0 | AAACCCACAAGTCATC-1-0 | AAACCCACACACACTA-1-0 | AAACCCACACCCATAA-1-0 | AAACCCACACGCACCA-1-0 | AAACCCACACTGCGAC-1-0 | AAACCCACAGGACTTT-1-0 | AAACCCAGTTGGGTTT-1-0 | AAACCCATCAGGTAAA-1-0 | ⋯ | TTTGTTGCAATCCTTT-1-2 | TTTGTTGCATGACTAC-1-2 | TTTGTTGGTCAAGTTC-1-2 | TTTGTTGGTGGATTTC-1-2 | TTTGTTGGTTACAGCT-1-2 | TTTGTTGGTTCTTGTT-1-2 | TTTGTTGGTTGCCGAC-1-2 | TTTGTTGTCCGGTAGC-1-2 | TTTGTTGTCTAGAACC-1-2 | TTTGTTGTCTTCTGGC-1-2 |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| NOC2L | 1.0012 | 0.8496507 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | ⋯ | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 0.8995852 | 0.8588233 | 1.0012 | 1.0012 | 1.0012 | 1.0012 |\n",
       "| ISG15 | 1.0012 | 0.8509542 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | ⋯ | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 0.8996055 | 0.8595151 | 1.0012 | 1.0012 | 1.0012 | 1.0012 |\n",
       "| TNFRSF4 | 1.0012 | 0.8522806 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | ⋯ | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 0.8997827 | 0.8602901 | 1.0012 | 1.0012 | 1.0012 | 1.0012 |\n",
       "| SDF4 | 1.0012 | 0.8534545 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | ⋯ | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 0.8999158 | 0.8609641 | 1.0012 | 1.0012 | 1.0012 | 1.0012 |\n",
       "| B3GALT6 | 1.0012 | 0.8546630 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | ⋯ | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 0.9001891 | 0.8617201 | 1.0012 | 1.0012 | 1.0012 | 1.0012 |\n",
       "| UBE2J2 | 1.0012 | 0.8557378 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 1.0012 | ⋯ | 1.0012 | 1.0012 | 1.0012 | 1.0012 | 0.9004141 | 0.8623833 | 1.0012 | 1.0012 | 1.0012 | 1.0012 |\n",
       "\n"
      ],
      "text/plain": [
       "        AAACCCAAGCTAATGA-1-0 AAACCCAAGTATGTAG-1-0 AAACCCACAAGTCATC-1-0\n",
       "NOC2L   1.0012               0.8496507            1.0012              \n",
       "ISG15   1.0012               0.8509542            1.0012              \n",
       "TNFRSF4 1.0012               0.8522806            1.0012              \n",
       "SDF4    1.0012               0.8534545            1.0012              \n",
       "B3GALT6 1.0012               0.8546630            1.0012              \n",
       "UBE2J2  1.0012               0.8557378            1.0012              \n",
       "        AAACCCACACACACTA-1-0 AAACCCACACCCATAA-1-0 AAACCCACACGCACCA-1-0\n",
       "NOC2L   1.0012               1.0012               1.0012              \n",
       "ISG15   1.0012               1.0012               1.0012              \n",
       "TNFRSF4 1.0012               1.0012               1.0012              \n",
       "SDF4    1.0012               1.0012               1.0012              \n",
       "B3GALT6 1.0012               1.0012               1.0012              \n",
       "UBE2J2  1.0012               1.0012               1.0012              \n",
       "        AAACCCACACTGCGAC-1-0 AAACCCACAGGACTTT-1-0 AAACCCAGTTGGGTTT-1-0\n",
       "NOC2L   1.0012               1.0012               1.0012              \n",
       "ISG15   1.0012               1.0012               1.0012              \n",
       "TNFRSF4 1.0012               1.0012               1.0012              \n",
       "SDF4    1.0012               1.0012               1.0012              \n",
       "B3GALT6 1.0012               1.0012               1.0012              \n",
       "UBE2J2  1.0012               1.0012               1.0012              \n",
       "        AAACCCATCAGGTAAA-1-0 ⋯ TTTGTTGCAATCCTTT-1-2 TTTGTTGCATGACTAC-1-2\n",
       "NOC2L   1.0012               ⋯ 1.0012               1.0012              \n",
       "ISG15   1.0012               ⋯ 1.0012               1.0012              \n",
       "TNFRSF4 1.0012               ⋯ 1.0012               1.0012              \n",
       "SDF4    1.0012               ⋯ 1.0012               1.0012              \n",
       "B3GALT6 1.0012               ⋯ 1.0012               1.0012              \n",
       "UBE2J2  1.0012               ⋯ 1.0012               1.0012              \n",
       "        TTTGTTGGTCAAGTTC-1-2 TTTGTTGGTGGATTTC-1-2 TTTGTTGGTTACAGCT-1-2\n",
       "NOC2L   1.0012               1.0012               0.8995852           \n",
       "ISG15   1.0012               1.0012               0.8996055           \n",
       "TNFRSF4 1.0012               1.0012               0.8997827           \n",
       "SDF4    1.0012               1.0012               0.8999158           \n",
       "B3GALT6 1.0012               1.0012               0.9001891           \n",
       "UBE2J2  1.0012               1.0012               0.9004141           \n",
       "        TTTGTTGGTTCTTGTT-1-2 TTTGTTGGTTGCCGAC-1-2 TTTGTTGTCCGGTAGC-1-2\n",
       "NOC2L   0.8588233            1.0012               1.0012              \n",
       "ISG15   0.8595151            1.0012               1.0012              \n",
       "TNFRSF4 0.8602901            1.0012               1.0012              \n",
       "SDF4    0.8609641            1.0012               1.0012              \n",
       "B3GALT6 0.8617201            1.0012               1.0012              \n",
       "UBE2J2  0.8623833            1.0012               1.0012              \n",
       "        TTTGTTGTCTAGAACC-1-2 TTTGTTGTCTTCTGGC-1-2\n",
       "NOC2L   1.0012               1.0012              \n",
       "ISG15   1.0012               1.0012              \n",
       "TNFRSF4 1.0012               1.0012              \n",
       "SDF4    1.0012               1.0012              \n",
       "B3GALT6 1.0012               1.0012              \n",
       "UBE2J2  1.0012               1.0012              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(infercnv_obj@expr.data)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "d0a11cbd",
   "metadata": {},
   "outputs": [],
   "source": [
    "cnvRes <- infercnv_obj@expr.data\n",
    "# write results\n",
    "write.table(\n",
    "cnvRes,\n",
    "file = stringr::str_interp('/data/inferCNVResults15Kcells/cnvResAllExpr.txt'),\n",
    "sep = '\\t',\n",
    "row.names = T,\n",
    "col.names = T,\n",
    "quote = F)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
